File: eunbi/week7/완주하지못한선수.java
package algorithmStudy;

import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<String, Integer>();
        
        for(String i : completion)
            map.put(i, map.getOrDefault(i, 0) + 1);
        
        for(String i : participant){
            if(map.containsKey(i)==false || map.get(i)==0){
                answer = i;
                break;
            }
            else
                map.put(i, map.get(i)-1);
        }
        
        
        return answer;
    }
}

File: SiHyeon/week7/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    multiset<string> participants; //연관 컨테이너인 set을 활용한다.
    for (auto p : participant)
        participants.insert(p);
    for (auto c : completion)
    {
        auto iter = participants.find(c);
        if (iter != participants.end()) //탐색에 성공하면 지워줌
            participants.erase(iter);
    }
    for (auto p : participants)
        answer += p;
    return answer;
}